go/archtest/runner_required_sentinel_test.go

```
@@ -0,0 +1,249 @@
+	}
+	// Restrict to the error constructors so an unrelated helper that merely takes
+	// such a string (e.g. a log call) is not flagged.
+	sel, ok := call.Fun.(*ast.SelectorExpr)
+	if !ok {
+		return false
+	}
+	pkg, ok := sel.X.(*ast.Ident)
+	if !ok {
+		return false
+	}
+	if !((pkg.Name == "errors" && sel.Sel.Name == "New") || (pkg.Name == "fmt" && sel.Sel.Name == "Errorf")) {
```

golangci-lint (2.12.2)

[error] QF1001: could apply De Morgan's law

(staticcheck)